src/main/java/com/odysseusinc/arachne/portal/service/ExternalLoginService.java

```diff
@@ -0,0 +1,84 @@
+package com.odysseusinc.arachne.portal.service;
+
+import com.odysseusinc.arachne.portal.model.ExternalLogin;
+import com.odysseusinc.arachne.portal.model.IUser;
+import com.odysseusinc.arachne.portal.repository.UserRepository;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.function.Supplier;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Service
+public class ExternalLoginService {
+    private static final Logger log = LoggerFactory.getLogger(ExternalLoginService.class);
+
+    /**
+     * Specifies whether newly created external users can be linked to existing local users.
+     */
+    @Value("${user.external.autolink:false}")
+    private boolean autolink;
+
+    /**
+     * Specifies whether new external users have their profile imported from authentication provider.
+     * Note that if autolink is enabled, it will be attempted first.
+     */
+    @Value("${user.external.autoimport:true}")
+    private boolean autoimport;
```

Comment:
If the flag is not used, it is better to remove it.

Reply:
Removed